core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/EntityConsistencyState.java

```diff
@@ -0,0 +1,81 @@
+package ai.timefold.solver.core.impl.domain.variable.declarative;
+
+import java.util.IdentityHashMap;
+
+import ai.timefold.solver.core.impl.domain.entity.descriptor.EntityDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.VariableDescriptor;
+
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@NullMarked
+public final class EntityConsistencyState<Solution_, Entity_> {
+    private final IdentityHashMap<Entity_, Boolean> entityToIsInconsistentMap;
+
+    @Nullable
+    private final ExternalizedShadowVariableInconsistentProcessor<Solution_> externalizedShadowVariableInconsistentProcessor;
+
+    private final VariableDescriptor<Solution_> randomDeclarativeVariableDescriptor;
```

Comment:
Consider how it's used, I suggest calling it `arbitrary` instead of `random`.

Also, I wonder... is there any way how consistency state would change without any declarative variable changing? If so, why would it ever be necessary to specifically trigger this? Consistency information is already set when the DSV value supplier is called. Therefore the update of consistency information will be guaranteed as a side-effect.

Reply:
`null` is a valid value for a declarative variable, so it could be the case all declarative variables on an entity remained `null` because all their suppliers returned `null`.